### PR TITLE
WINDUP-670: Make Rexster more convenient to use

### PR DESCRIPTION
**Pull request by mbriskar**
From now on, rexster is deployed by default everytime you debug within windup.

### CODE REVIEW

**Comment by mbriskar** [comment]
It will work after the furnace new version will be released with the patch.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/906/
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/275/
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/907/
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/276/
Test FAILed.

File: config-xml/tests/pom.xml [hunk 1]
```diff
@@ -47,15 +47,11 @@
...
-            <scope>test</scope>
-        </dependency>
-        <dependency>
-            <groupId>org.jboss.forge.furnace.test</groupId>
-            <artifactId>arquillian-furnace-classpath</artifactId>
-            <scope>test</scope>
-        </dependency>
+                <groupId>org.jboss.windup</groupId>
+                <artifactId>arquillian</artifactId>
+                <version>${project.version}</version>
```
Comment (mbriskar) [inline config-xml/tests/pom.xml]: note that this is not a forge addon

File: arquillian/addon/src/main/java/org/jboss/windup/arquillian/WindupFurnaceAddonDeploymentEnhancer.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+ * Deployment service that deploys the rexster addon everytime debugging is on.
+ *
+ * @author <a href="mailto:mbriskar@gmail.com">Matej Briškár</a>
+ */
+public class WindupFurnaceAddonDeploymentEnhancer implements AddonDeploymentScenarioEnhancer
+{
+    @Override public List<DeploymentDescription> enhance(TestClass testClass, List<DeploymentDescription> deployments)
+    {
+        String debugProperty = System.getProperty("maven.surefire.debug");
+        if(Boolean.getBoolean(debugProperty)) {
```
Comment (gastaldi) [inline arquillian/addon/src/main/java/org/jboss/windup/arquillian/WindupFurnaceAddonDeploymentEnhancer.java]: This should be if(Boolean.getBoolean("maven.surefire.debug"))

File: rexster/tests/src/test/java/org/jboss/windup/graph/rexster/test/RexsterDefaultDeploymentTest.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import org.junit.runner.RunWith;
+
+import java.io.IOException;
+import java.net.Socket;
+
+/**
+ * Tests that rexster is properly deployed thanks to Arquillian ad
+ */
+@RunWith(Arquillian.class)
+public class RexsterDefaultDeploymentTest
```
Comment (mbriskar) [inline rexster/tests/src/test/java/org/jboss/windup/graph/rexster/test/RexsterDefaultDeploymentTest.java]: This is the test testing that it is deployed when surefire debug is ON

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/923/
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/293/
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/924/
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/294/
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/926/
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/296/
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/927/
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/297/
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/928/
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/298/
Test FAILed.

**Comment by mbriskar** [comment]
retest please

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/930/
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/300/
Test FAILed.

File: config-xml/tests/pom.xml (hunk 1 above)
Comment (lincolnthree) [inline config-xml/tests/pom.xml]: maybe this should be renamed to "windup-test-harness" instead of "arquillian" ?

**Comment by mbriskar** [comment]
problem is on my side, fixing immediately

**Comment by mbriskar** [comment]
hopefully fixed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/934/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/304/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules:windup-rulesets' /><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/304/org.jboss.windup.rules$windup-rulesets/testReport'>org.jboss.windup.rules:windup-rulesets</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/304/org.jboss.windup.rules$windup-rulesets/testReport/org.jboss.windup.rules.tests/WindupRulesTest/testWindupRules/'><strong>org.jboss.windup.rules.tests.WindupRulesTest.testWindupRules</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/935/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/305/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules:windup-rulesets' /><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/305/org.jboss.windup.rules$windup-rulesets/testReport'>org.jboss.windup.rules:windup-rulesets</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/305/org.jboss.windup.rules$windup-rulesets/testReport/org.jboss.windup.rules.tests/WindupRulesTest/testWindupRules/'><strong>org.jboss.windup.rules.tests.WindupRulesTest.testWindupRules</strong></a></li></ul>
Test FAILed.

File: config-groovy/tests/pom.xml [hunk 4]
```diff
@@ -39,14 +39,10 @@
...
-            <artifactId>furnace-test-harness</artifactId>
-            <scope>test</scope>
-        </dependency>
-        <dependency>
-            <groupId>org.jboss.forge.furnace.test</groupId>
-            <artifactId>arquillian-furnace-classpath</artifactId>
-            <scope>test</scope>
-        </dependency>
+                <groupId>org.jboss.windup</groupId>
+                <artifactId>windup-test-harness</artifactId>
```
Comment (lincolnthree) [inline config-groovy/tests/pom.xml]: good.

File: windup-test-harness/pom.xml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    <artifactId>windup-test-harness</artifactId>
+    
+
+    <name>Windup Engine - Arquillian extension</name>
+    <description>Used to register extensions for the furnace arquillian module so that it is possible to deploy additional
+    addons by default.</description>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.jboss.shrinkwrap</groupId>
```
Comment (lincolnthree) [inline windup-test-harness/pom.xml]: Is this really needed?

**Comment by mbriskar** [comment]
what about the failing tests in rulesets? Or didn't this PR cause it?

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/938/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/308/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules:windup-rulesets' /><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/308/org.jboss.windup.rules$windup-rulesets/testReport'>org.jboss.windup.rules:windup-rulesets</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/308/org.jboss.windup.rules$windup-rulesets/testReport/org.jboss.windup.rules.tests/WindupRulesTest/testWindupRules/'><strong>org.jboss.windup.rules.tests.WindupRulesTest.testWindupRules</strong></a></li></ul>
Test FAILed.

**Comment by lincolnthree** [comment]
I don't think this PR caused it.

**Pull request merged**